service/history/chasm/components/activity/activity.go

```diff
@@ -0,0 +1,168 @@
+package activity
+
+import (
+	"time"
+
+	"go.temporal.io/api/common/v1"
+	"go.temporal.io/server/api/matchingservice/v1"
+	persistencepb "go.temporal.io/server/api/persistence/v1"
+	"go.temporal.io/server/service/history/chasm"
+	"google.golang.org/protobuf/types/known/timestamppb"
+)
+
+type (
+	Activity struct {
+		// ALL FIELDS MUST BE EXPORTED for reflection to work
+
+		// In V1, we will only support only one non-chasm.XXX field in the struct.
+		// and that field must be a proto.Message.
+		// TODO: define a serializer/deserializer interface?
+		//
+		// Framework will try to recognize the type and do serialization/deserialization
+		// proto.Message is recommended so the component get compatibility if state definition changes
+		State persistencepb.ActivityInfo // proto.Message
```

Comment:
Let's call this standard field `Data`, since `State` is usually an enum.

Reply:
Alternatively, we can let you embed a proto and treat that as the data field. That could be slightly nicer to use.

Reply:
Agree with naming change. Disagree with proto embed. I'd prefer for it to just be explicit.